src/SingleCellAnnotator.imjoy.html

```diff
@@ -847,9 +838,40 @@
     },
     startRedrawAnnotation() {
       this.current_image_data.mask_adjusted = true;
-      this.vector_source.clear()
+      if (this.current_task_name !== 'segmentation'){
+        this.vector_source.clear()
+      };
       this.startMaskAnnotation(this.vector_source, this.freehand_mode)
     },
+    // to remove selected annotation
+    async deleteAnnotation() {
+      if (this.vector_source) {
+        const features = this.select.getFeatures()
+        if(features.getLength()>0){
+            features.forEach((feature)=>{
+                console.log('removing ', feature)
+                this.vector_source.removeFeature(feature);
+                this.select.getFeatures().remove(feature);
+            });
+        }
+        else{
+            try {
+                const answer = await api.confirm({content: "Do you really want to delete the annotations?", confirm_text: 'Yes, delete it.'})
+                if (answer) {
+                  const features = this.vector_source.getFeatures();
+                  features.forEach((feature) => {
+                    const ann = this.annotation_types[feature.get('label')]
+                    this.vector_source.removeFeature(feature);
+                  });
+                  this.undo_button_flag = false
```

Comment:
Do we have an undo button?

Reply:
no undo button so far

Reply:
will add

Reply:
I think we can do without an undo for now.